spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/InboundFileSynchronizer.java

```diff
@@ -22,10 +22,23 @@
  * Strategy for synchronizing from a remote File system to a local directory.
  *
  * @author Mark Fisher
+ * @author Gary Russell
  * @since 2.0
  */
 public interface InboundFileSynchronizer {
 
+	/**
+	 * Synchronize all available files to the local directory;
+	 * @param localDirectory the directory.
+	 */
 	void synchronizeToLocalDirectory(File localDirectory);
 
+	/**
+	 * Synchronize up to maxFetchSize files to the local directory;
+	 * @param localDirectory the directory.
+	 */
+	default void synchronizeToLocalDirectory(File localDirectory, int maxFetchSize) {
```

Comment:
Great! :smile: 
I guess in addition this interface can be marked with `FunctionalInterface`. Although not sure how that helps...